src/components/shared/TreeView.js

```diff
@@ -581,6 +583,9 @@ class TreeView<
             specialItems={this._specialItems}
             disableOverscan={!!disableOverscan}
             onCopy={this._onCopy}
+            // If there is a deep call node depth, expand the width, or else keep it
+            // at 3000 wide.
+            containerWidth={Math.max(3000, maxNodeDepth * 10 + 2000)}
```

Comment:
Mabe use the formula alone here, but add a min-width in CSS ? Could be easier to tune ?
Although maybe you prefer to have it all in one place.